adapters/projects/polaris/lp_transfer.go

```diff
@@ -0,0 +1,197 @@
+package polaris
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/taikoxyz/trailblazer-adapters/adapters"
+	"github.com/taikoxyz/trailblazer-adapters/adapters/contracts/balancer_vault"
+	"github.com/taikoxyz/trailblazer-adapters/adapters/contracts/erc20"
+)
+
+const (
+	VaultAddress             = "0x723F40720836a03f1F20441FbDa3D109fc99640F" // Replace with the actual vault contract address
+	logPoolBalanceChangedSig = "PoolBalanceChanged(bytes32,address,address[],int256[],uint256[])"
+)
+
+type LPTransferIndexer struct {
+	client    *ethclient.Client
+	addresses []common.Address
+}
+
+func NewLPTransferIndexer(client *ethclient.Client, addresses []common.Address) *LPTransferIndexer {
+	return &LPTransferIndexer{
+		client:    client,
+		addresses: addresses,
+	}
+}
+
+var _ adapters.LogIndexer[adapters.LPTransfer] = &LPTransferIndexer{}
+
+func (indexer *LPTransferIndexer) Addresses() []common.Address {
+	return indexer.addresses
+}
+
+func (indexer *LPTransferIndexer) Index(ctx context.Context, logs ...types.Log) ([]adapters.LPTransfer, error) {
```

Comment:
Would you mind, adding a test case for your indexer?

Reply:
I've added a test case for the indexer to ensure it performs as expected.